src/ota.rs

```diff
@@ -161,7 +164,7 @@ impl EspOtaUpdate {
 }
 
 #[derive(Debug)]
-pub struct EspOta(EspOtaUpdate);
+pub struct EspOta(Option<EspOtaUpdate>);
```

Comment:
Can you elaborate why introducing `Option` here is necessary? It increases the payload of the `EspOta` struct (not that this is the end of the world), yet I'm not sure what problem it is addressing?